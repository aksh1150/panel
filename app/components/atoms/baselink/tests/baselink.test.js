import React from 'react';
import { cleanup, render, fireEvent } from 'react-testing-library';
import 'jest-dom/extend-expect';

import BaseLink from '../index';
afterEach(cleanup);
describe('<BaseLink />', () => {
  it('should render and match the snapshot', () => {
    const { asFragment } = render(
      <BaseLink className target to="home">
        Home
      </BaseLink>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('has a class', () => {
    const { getByTestId } = render(<BaseLink className>Home</BaseLink>);
    expect(getByTestId('id')).toHaveClass('baselink');
  });

  it('has clicked', () => {
    const { getByTestId } = render(
      <BaseLink href="https://google.com">Google</BaseLink>,
    );
    fireEvent.click(getByTestId('id'));
  });
  it('has anchor attribute ', () => {
    const { getByTestId } = render(
      <BaseLink link="https://google.com">Google</BaseLink>,
    );
    expect(getByTestId('id')).toHaveAttribute('href', 'https://google.com');
  });
});
