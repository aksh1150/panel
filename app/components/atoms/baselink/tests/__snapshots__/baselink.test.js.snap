// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseLink /> should render and match the snapshot 1`] = `
<DocumentFragment>
  <a
    class="baselink true"
    data-testid="id"
    rel="noreferrer"
    target="_blank"
  >
    Home
  </a>
</DocumentFragment>
`;
